libstuff/SQResult.cpp

```diff
@@ -1,6 +1,63 @@
 #include <libstuff/libstuff.h>
 #include "SQResult.h"
 
+SQResultRow::SQResultRow(SQResult& result, size_t count) : vector<string>(count), result(&result) {
+}
+
+SQResultRow::SQResultRow() : vector<string>(), result(nullptr) {
+}
+
+void SQResultRow::push_back(const string& s) {
+    vector<string>::push_back(s);
+}
+
+string& SQResultRow::operator[](const size_t& key) {
+    return vector<string>::operator[](key);
+}
+const string& SQResultRow::operator[](const size_t& key) const {
+    return vector<string>::operator[](key);
+}
+
+SQResultRow& SQResultRow::operator=(const SQResultRow& other) {
+    vector<string>::operator=(other);
+    result = other.result;
+    return *this;
+}
+
+string& SQResultRow::operator[](const string& key) {
+    if (result) {
+        for (size_t i = 0; i < result->headers.size(); i++) {
+
+            // If the headers have more entries than the row (they really shouldn't), break early instead of segfaulting.
+            if (i >= size()) {
+                break;
+            }
+
+            if (result->headers[i] == key) {
+                return (*this)[i];
+            }
+        }
+    }
+    throw out_of_range("No column named " + key);
```

Comment:
Should we move this to a private method so we don't have to duplicate the code twice?

Reply:
It doesn't work in this case because of how the templated code works, one calls `const` base class methods and one calls non-`const` base class methods.